src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Services/DataSetVersionService.cs

```diff
@@ -23,31 +23,50 @@ internal class DataSetVersionService(
     IDataSetVersionPathResolver dataSetVersionPathResolver
 ) : IDataSetVersionService
 {
+    public async Task<Either<ActionResult, (Guid dataSetId, Guid dataSetVersionId)>> CreateInitialVersion(
```

Comment:
Does this need to return the `dataSetId` as a tuple? This method already receives a `dataSetId` as a parameter so we already know what this is in the consumer.

Would simplify this by changing the return type to `Task<Either<ActionResult, Guid>>`